mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -117,6 +118,7 @@ class RowMatrix @Since("1.0.0") (
     // Computes n*(n+1)/2, avoiding overflow in the multiplication.
     // This succeeds when n <= 65535, which is checked above
     val nt = if (n % 2 == 0) ((n / 2) * (n + 1)) else (n * ((n + 1) / 2))
+    val grammianSizeInMb = (SizeEstimator.estimate(new BDV[Double](nt)) / 1000).toInt
```

Comment:
I'm not sure if this is the right way to do it here. I was hesitating between:
- Doing that, meaning using a standard way of estimating object size, at the cost of allocating the dense vector
- Not allocating the dense vector, at the cost of estimating its size with an ad-hoc computation

Reply:
I wouldn't allocate a potentially large vector just for this. Just go with the ad-hoc estimate of `nt * 8 / 1000000`
Nit: grammian -> gramian

Reply:
Fine, I'll change that (and correct the typo in gramian)